pkg/materialize/source_load_generator.go

```diff
@@ -10,15 +10,31 @@ type TableLoadgen struct {
 	Alias string
 }
 
+type CounterOptions struct {
+	TickInterval   string
+	MaxCardinality bool
+}
```

Comment:
Indeed this is nontrivial. I usually use the testdrive for some references but in this case it does not cover all of the cross options: 
https://github.com/MaterializeInc/materialize/blob/main/test/testdrive/load-generator.td

For example with the counter you can actually specify a scale factor: `Expected one of SCALE or TICK or MAX`, but in the docs we only explicitly mention the TPCH loadgen.

I think that the table that you've shared is spot on!